packages/database-typeorm/__tests__/typeorm-password.ts

```diff
@@ -0,0 +1,467 @@
+
+    it('should throw if email is not found', async () => {
+      const accountsTypeorm = new AccountsTypeorm({ connection });
+      const userId = await accountsTypeorm.createUser(user);
+      await expect(accountsTypeorm.verifyEmail(userId, 'unknowemail')).rejects.toThrowError(
+        'Email not found'
+      );
+    });
+
+    it('should verify email', async () => {
+      const accountsTypeorm = new AccountsTypeorm({ connection });
+      const userId = await accountsTypeorm.createUser(user);
+      let retUser = await accountsTypeorm.findUserById(userId);
+      expect(retUser!.emails!.length).toEqual(1);
+      expect(retUser!.emails![0].address).toBe(user.email);
+      expect(retUser!.emails![0].verified).toBe(false);
+      await accountsTypeorm.verifyEmail(userId, user.email);
+      retUser = await accountsTypeorm.findUserById(userId);
+      expect(retUser!.emails!.length).toEqual(1);
+      expect(retUser!.emails![0].address).toBe(user.email);
+      expect(retUser!.emails![0].verified).toBe(true);
+      expect(retUser!.createdAt).not.toEqual(retUser!.updatedAt);
+    });
+  });
+
+  describe('setUsername', () => {
+    it('should throw if user is not found', async () => {
+      const accountsTypeorm = new AccountsTypeorm({ connection });
+      await expect(
+        accountsTypeorm.setUsername('402fcb56-e325-4950-9166-63855da0a3fe', 'username')
+      ).rejects.toThrowError('User not found');
+    });
+
+    it('should change username', async () => {
+      const accountsTypeorm = new AccountsTypeorm({ connection });
+      const username = 'johnsdoe';
+      const userId = await accountsTypeorm.createUser(user);
+      await accountsTypeorm.setUsername(userId, username);
+      const retUser = await accountsTypeorm.findUserById(userId);
+      expect(retUser!.username).toEqual(username);
+      expect(retUser!.createdAt).not.toEqual(retUser!.updatedAt);
+    });
+  });
+
+  describe('setPassword', () => {
+    it('should throw if user is not found', async () => {
+      const accountsTypeorm = new AccountsTypeorm({ connection });
+      await expect(
+        accountsTypeorm.setPassword('402fcb56-e325-4950-9166-63855da0a3fe', 'toto')
+      ).rejects.toThrowError('User not found');
+    });
+
+    it('should change password', async () => {
+      const accountsTypeorm = new AccountsTypeorm({ connection });
+      const newPassword = 'newpass';
+      const userId = await accountsTypeorm.createUser(user);
+      await accountsTypeorm.setPassword(userId, newPassword);
+      const retUser = await accountsTypeorm.findUserById(userId);
+      const services: any = retUser!.services;
+      expect(services.password[0].bcrypt).toBeTruthy();
+      expect(services.password[0].bcrypt).toEqual(newPassword);
+      expect(retUser!.createdAt).not.toEqual(retUser!.updatedAt);
+    });
+  });
+
+  describe('setUserDeactivated', () => {
+    // TODO: This should be added
+    // it('should throw if user is not found', async () => {
```

Comment:
Should this be done before merging?

Reply:
I actually added these tests because there is another bug I found in typeorm (will create an issue and PR for it) and the thought of making changes without any tests was scary. I wanted this PR to be only tests with no code changes whatsoever so that I can be sure things don't break. In the course of adding these tests from mongo, I also found a few other inconsistencies between the APIs. Will file bug reports for those as well.

TL;DR No, I want this to be just tests and will create separate PRs for other fixes. That commented out code is a reminder for me